src/canisters/frontend/ic-certified-assets/src/state_machine.rs

```diff
@@ -555,6 +555,33 @@ impl State {
             token: create_token(asset, &content_encoding, enc, &key, chunk_index),
         })
     }
+
+    pub fn get_asset_properties(&self, key: Key) -> Result<AssetProperties, String> {
+        let asset = self
+            .assets
+            .get(&key)
+            .ok_or_else(|| "asset not found".to_string())?;
+
+        Ok(AssetProperties {
+            max_age: asset.max_age,
+            headers: asset.headers.clone(),
+        })
+    }
+
+    pub fn set_asset_properties(&mut self, arg: SetAssetPropertiesArguments) -> Result<(), String> {
```

Comment:
Please add state machine tests for these.  The state machine tests should go into as much detail as possible.  For example:
- what happens if you pass Some for headers and None for max_age
- have a test that demonstrates that after you've set either to Some(), you cannot change either back to None (and: is this desired?)